### PR TITLE
#1014 Fix DefaultViewLocator

### PR DESCRIPTION
**Pull request by ChrisPulman**
**What kind of change does this PR introduce? (Bug fix, feature, docs update, ...)**
#1014 bug in DefaultViewLocator

**What is the current behavior? (You can also link to an open issue here)**
Fails to resolve IViewFor<INTERFACE> i.e. A concrete class with an interface registered in Splat
MainViewModel as IMainViewModel register a view as IViewFor<IMainViewModel>
the DefaultViewLocator fails to resolve the view
**What is the new behavior (if this is a feature change)?**
Resolves IVeiwFor<INTERFACE>

**Does this PR introduce a breaking change?**
No, unless desired operation is to fail when using interfaces as IViewFor registration

**Please check if the PR fulfills these requirements**
- [ ] The commit follows our guidelines: https://github.com/reactiveui/reactiveui#contribute
- [ ] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features)

**Other information**:

### CODE REVIEW

File: src/ReactiveUI-Blend.nuspec [hunk 1]
```diff
@@ -23,11 +23,11 @@
       </group>
       <group targetFramework="uap10.0">
         <dependency id="reactiveui-core" version="$version$" />
-        <dependency id="System.Diagnostics.Debug" version="4.0.10" />
         <dependency id="Microsoft.Xaml.Behaviors.Uwp.Managed" version="1.1.0" />
         <dependency id="Rx-Core" version="2.2.5" />
         <dependency id="Rx-Interfaces" version="2.2.5" />
         <dependency id="Rx-Linq" version="2.2.5" />
+        <dependency id="System.Diagnostics.Debug" version="4.0.10" />
```
Comment (ChrisPulman) [inline src/ReactiveUI-Blend.nuspec]: @kentcb  This is purely an alphabetical sorting that happened when I saved the file, the previous also had this same dependency just at the top of the file.

Comment (ChrisPulman) [inline src/ReactiveUI-Blend.nuspec]: @kentcb I had modified the existing test that was for this to use an interface as opposed to a var
I made 4 commits on this issue, sorry to confuse things but Appveyor was complaining and I wasn't sure that I had satisfied all the requirements.
I am about to fly to China I will look to see if I can come up with some additional Tests to better prove the operation of the ViewLocator whilst there.
I hope this doesn't hold back the ability to merge this commit.

**Comment by kentcb** [comment]
@ChrisPulman If you merge latest upstream rxui7-master branch into your branch, you should get AppVeyor to build fine.

**Comment by kentcb** [comment]
@ChrisPulman  If you merge latest upstream rxui7-master branch into your branch, you should get AppVeyor to build fine.

**Comment by kentcb** [comment]
@ChrisPulman I know there are currently no tests for `DefaultViewLocator`, but I think it's likely an oversight. #1014 includes at least one test. Any chance you could add it and - if you're feeling really generous - maybe supplement with your own?

**Comment by ChrisPulman** [comment]
@kentcb I have updated the branch and all appveyor checks passed, I am going to get on the Plane now. I hope this is ok and what you had requested. Thanks.

**Comment by kentcb** [comment]
@ChrisPulman can you please allow edits from maintainers on this PR:

image

That way, I can fix the build and add tests before merging.

**Comment by ChrisPulman** [comment]
@kentcb I have updated as requested.

**Comment by kentcb** [comment]
Just tried pushing to this branch but didn't have much luck. Might just open a separate PR if I can't figure this out...

**Comment by ChrisPulman** [comment]
looking at the AppVeyor log it looks like #1259 merge is the root cause of the issues I am not sure exactly what the issue is though:
Error: NuGet: Process returned an error (exit code 1).
Command exited with code 1

**Comment by kentcb** [comment]
Hmm, I don't think that's merged yet though (it's also failing, but completely different reason). And there's a couple of other branches failing for other reasons. This CI is great when it works, but I feel completely helpless/out of my depth when it doesn't.

**Comment by ChrisPulman** [comment]
I have synced the repo and at this moment I have 9 projects that wont build, I am trying to find out why, it looks like missing references, this could possibly be the cause for failing especially if you are using VS and have had a successful build before, closed your solution and then opened it again, it will automatically find the dll from your builds, its worth checking the paths or delete your obj and bin dir's see if it builds then.

**Comment by kentcb** [comment]
To be clear, are you trying to build this branch or another?

**Comment by ChrisPulman** [comment]
I am using this branch, after executing sync in Git for Windows

**Comment by kentcb** [comment]
Weird. I just switched to this branch, did a `git clean -xdf`, a `git pull upstream`, and then tried building in VS. It worked fine (some warnings, but that was it).

**Comment by ChrisPulman** [comment]
Thnaks for doing that, I will keep trying to get a clean build, can I suggest adding the nuget package System.Windows.Interactivity.WPF to the ReactiveUI.Blend_Net45 project to ensure that the reference is common for all, I have had so many issues with the blend libs being different or even missing on some machines, I guess mine is one of those, mine being a different version :(
The other issues seam to be related to the Andriod and WinRT projects I am thinking my VS install has issues or the wrong versions of android

**Comment by ChrisPulman** [comment]
C:\Program Files (x86)\Microsoft SDKs\Windows\v8.1\ExtensionSDKs\BehaviorsXamlSDKManaged\
is missing on my machine hence issues with WinRT. Other than this all non Android projects build. 

What SDK should be being used for Android?

**Comment by kentcb** [comment]
Superseded by #1282.

**Pull request closed without merging**